docs/tutorials/python/sagas/index.md

```diff
@@ -0,0 +1,586 @@
+If the booking process is cancelled, the response indicates this. Otherwise, it provides details about the booked car, hotel, and flight.
+
+Now to start the Client, run the following command in your new terminal:
+
+```bash
+# terminal two
+python3 run_workflow.py
+```
+
+Once the Client is set up, you can start the booking process and see the Saga pattern in action.
+
+## Start the booking process
+
+To run the booking process, you can use the following `curl` command to send a `POST` request to the `/book` endpoint.
+This request will trigger the Workflow, and you will receive a response with the booking details or a cancellation message.
+
+```bash
+# terminal three
+curl -X POST http://localhost:3002/book \
+-H "Content-Type: application/json" \
+-d '{
+    "name": "John Doe",
+    "attempts": 5,
+    "car": "valid-car-id",
+    "hotel": "valid-hotel-id",
+    "flight": "valid-flight-id"
+}'
+```
+
+
+You'll see a JSON response similar to the following:
+
+```json
+{
+  "cancelled": false,
+  "car": "Car: valid-car-id",
+  "flight": "Flight: valid-flight-id",
+  "hotel": "Hotel: valid-hotel-id",
+  "result": "Booked car: valid-car-id Booked hotel: valid-hotel-id Booked flight: valid-flight-id",
+  "user_id": "john-doe-184942"
+}
+```
+
+
+You've successfully initiated and completed your booking process using the Saga pattern with Temporal in Python.
+Next, you'll learn how to simulate errors to test the robustness of your implementation.
+
+## Simulate an error
+
+To ensure your implementation can handle failures gracefully, you will simulate a booking failure.
+This step will demonstrate how the Saga pattern with Temporal manages to roll back in case of errors.
+
+To simulate a booking failure, you can use the following `curl` command.
+This request includes an invalid hotel booking ID, which will cause the booking process to fail and trigger the rollback process.
+
+```bash
+# terminal three
+curl -X POST http://localhost:3002/book \
+-H "Content-Type: application/json" \
+-d '{
+    "name": "Jane Smith",
+    "attempts": 3,
+    "car": "valid-car-id",
+    "hotel": "invalid-hotel-id",
+    "flight": "valid-flight-id"
+}'
+```
+
+The value `invalid` will trigger an exception, causing the booking to rollback.
+
+
+The output in your terminal will be a JSON response similar to the following:
+
+```json
+{
+  "cancelled": true,
+  "result": "Voyage cancelled",
+  "user_id": "jane-smith-609592"
+}
+```
+
+In this case, the booking process was cancelled due to the invalid hotel booking ID.
+The Saga pattern ensures that any completed bookings are rolled back, maintaining a consistent state.
+
+To verify the actions that took place during the booking process, you can use the Temporal Web UI.
+
+Open your instance of the Temporal Web UI in your browser and navigate to the **Workflows** tab.
+
+:::note View the Web UI
+
+If you are using the Temporal CLI, you can visit [localhost:8233/](http://localhost:8233/).
```

Comment:
Just a note that @MasonEgger told me that the learn site universally uses 8080 instead of 8233